src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_pix2pix_zero.py

```diff
@@ -0,0 +1,803 @@
+# Copyright 2023 Pix2Pix Zero Authors and The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import PIL
+import torch
+from transformers import (
+    AutoProcessor,
+    BlipForConditionalGeneration,
+    CLIPFeatureExtractor,
+    CLIPTextModel,
+    CLIPTokenizer,
+)
+
+from ...models import AutoencoderKL, UNet2DConditionModel
+from ...models.cross_attention import CrossAttention
+from ...schedulers import KarrasDiffusionSchedulers
+from ...utils import is_accelerate_available, logging, randn_tensor
+from ..pipeline_utils import DiffusionPipeline
+from . import StableDiffusionPipelineOutput
+from .safety_checker import StableDiffusionSafetyChecker
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+def generate_caption(image, captioner, processor, return_image=True):
+    """Generates caption for a given image."""
+    inputs = processor(images=image, return_tensors="pt")
+    outputs = captioner.generate(inputs)
+    caption = processor.batch_deocde(outputs, skip_special_tokens=True)[0]
+    if return_image:
+        return caption, inputs["pixel_values"]
+    else:
+        return caption
+
+
+def prepare_unet(unet: UNet2DConditionModel):
+    """Modifies the UNet (`unet`) to perform Pix2Pix Zero optimizations."""
+    pix2pix_zero_attn_procs = {}
+    for name in unet.attn_processors.keys():
+        module_name = name.replace(".processor", "")
+        module = unet.get_submodule(module_name)
+        if "attn2" in name:
+            pix2pix_zero_attn_procs[name] = Pix2PixZeroCrossAttnProcessor(is_pix2pix_zero=True)
+            module.requires_grad_(True)
+        else:
+            pix2pix_zero_attn_procs[name] = Pix2PixZeroCrossAttnProcessor(is_pix2pix_zero=False)
```

Comment:
Cool that works! Yeah I was worried that you would get this error message (that the original `CrossAttnProcessor` doesn't accept the timesteps. Think this is a very nice approach. Alternatively I would have proposed to check here: https://github.com/huggingface/diffusers/blob/e55687e1e15407f60f32242027b7bb8170e58266/src/diffusers/models/cross_attention.py#L202 the signature of the processor and pop kwargs that are not accepted, but I think the approach you currently have here is nicer, so in favor of leaving as is :-)

Reply:
Yeah, I had thought of that but felt let's not touch the canonical code for now hahaha.